test/unit/connect-inject-deployment.bats

```diff
@@ -409,3 +409,142 @@ load _helpers
+  cd `chart_dir`
+  local object=$(helm template \
+      -x templates/connect-inject-deployment.yaml  \
+      --set 'connectInject.enabled=true' \
+      --set 'global.consulNamespacesEnabled=true' \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.containers[0].command' | tee /dev/stderr)
+  local actual=$(echo $object |
+    yq 'any(contains("enable-namespaces=true"))' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+  local actual=$(echo $object |
+    yq 'any(contains("consul-namespace=default"))' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+  local actual=$(echo $object |
+    yq 'any(contains("enable-namespace-mirroring"))' | tee /dev/stderr)
+  [ "${actual}" = "false" ]
+  local actual=$(echo $object |
+    yq 'any(contains("mirroring-prefix"))' | tee /dev/stderr)
+  [ "${actual}" = "false" ]
+}
+@test "connectInject/Deployment: mirroring options set with .connectInject.consulNamespaces.mirrorK8S=true" {
+  cd `chart_dir`
+  local object=$(helm template \
+      -x templates/connect-inject-deployment.yaml  \
+      --set 'connectInject.enabled=true' \
+      --set 'global.consulNamespacesEnabled=true' \
+      --set 'connectInject.consulNamespaces.mirrorK8S=true' \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.containers[0].command' | tee /dev/stderr)
+  local actual=$(echo $object |
+    yq 'any(contains("enable-namespaces=true"))' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+  local actual=$(echo $object |
+    yq 'any(contains("consul-namespace=default"))' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+  local actual=$(echo $object |
+    yq 'any(contains("enable-namespace-mirroring=true"))' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+  local actual=$(echo $object |
+    yq 'any(contains("mirroring-prefix"))' | tee /dev/stderr)
+  [ "${actual}" = "false" ]
+}
+@test "connectInject/Deployment: prefix can be set with .connectInject.consulNamespaces.mirroringPrefix" {
+  cd `chart_dir`
+  local object=$(helm template \
+      -x templates/connect-inject-deployment.yaml  \
+      --set 'connectInject.enabled=true' \
+      --set 'global.consulNamespacesEnabled=true' \
+      --set 'connectInject.consulNamespaces.mirrorK8S=true' \
+      --set 'connectInject.consulNamespaces.mirroringPrefix=k8s-' \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.containers[0].command' | tee /dev/stderr)
+  local actual=$(echo $object |
+    yq 'any(contains("enable-namespaces=true"))' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+  local actual=$(echo $object |
+    yq 'any(contains("consul-namespace=default"))' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+  local actual=$(echo $object |
+    yq 'any(contains("enable-namespace-mirroring=true"))' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+  local actual=$(echo $object |
+    yq 'any(contains("mirroring-prefix=k8s-"))' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+}
```

Comment:
Needs tests for logic around `CONSUL_HTTP_TOKEN` and `injector-acl-init` init container.

Reply:
Added.